pkg/ui/react-app/src/App.tsx

```diff
@@ -19,36 +19,67 @@ const defaultRouteConfig: { [component: string]: string } = {
   store: '/loaded',
 };
 
-const App: FC<PathPrefixProps & ThanosComponentProps> = ({ pathPrefix, thanosComponent }) => {
+const App: FC<ThanosComponentProps> = ({ thanosComponent }) => {
+  // This dynamically/generically determines the pathPrefix by stripping the first known
+  // endpoint suffix from the window location path. It works out of the box for both direct
+  // hosting and reverse proxy deployments with no additional configurations required.
+  let basePath = window.location.pathname;
+  const paths = [
+    '/graph',
+    '/alerts',
+    '/status',
+    '/tsdb-status',
+    '/flags',
+    '/config',
+    '/rules',
+    '/targets',
+    '/service-discovery',
+    '/stores',
+    '/blocks',
+    '/loaded',
+  ];
+  if (basePath.endsWith('/')) {
+    basePath = basePath.slice(0, -1);
+  }
+  if (basePath.length > 1) {
+    for (let i = 0; i < paths.length; i++) {
+      if (basePath.endsWith(paths[i])) {
+        basePath = basePath.slice(0, basePath.length - paths[i].length);
+        break;
+      }
+    }
+  }
+
   return (
     <ErrorBoundary>
-      <Navigation
-        pathPrefix={pathPrefix}
-        thanosComponent={thanosComponent}
-        defaultRoute={defaultRouteConfig[thanosComponent]}
-      />
-      <Container fluid style={{ paddingTop: 70 }}>
-        <QueryParamProvider reachHistory={globalHistory}>
-          <Router basepath={`${pathPrefix}/new`}>
-            <Redirect from="/" to={`${pathPrefix}/new${defaultRouteConfig[thanosComponent]}`} />
+      <PathPrefixContext.Provider value={basePath}>
```

Comment:
The CI is failing because the code is not properly formatted/linted. Try enabling auto formatting/linting in your editor or manually format all these files so that the code stays consistent (tip: you can run `make react-app-lint-fix` from the root of this repo or `yarn lint` from the `react-app` directory to auto-format everything).